08.ls_object/ls_file.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'etc'
+
+class LsFile
+  DIRECTORY_NUMBER = '40'
+
+  PERMISSION_MAP = {
+    '0' => '---',
+    '1' => '--x',
+    '2' => '-w-',
+    '3' => '-wx',
+    '4' => 'r--',
+    '5' => 'r-x',
+    '6' => 'rw-',
+    '7' => 'rwx'
+  }.freeze
+
+  TYPE_MAP = {
+    '01' => 'p',
+    '02' => 'c',
+    '40' => 'd',
+    '06' => 'b',
+    '10' => '-',
+    '12' => 'l',
+    '14' => 's'
+  }.freeze
+
+  attr_reader :name, :file_stat
+
+  def initialize(file_name)
+    @name = file_name
+    @file_stat = File.stat(file_name)
+  end
+
+  def type
+    TYPE_MAP[@file_stat.mode.to_s(8).slice(0..1)]
+  end
+
+  def permission
+    file_mode = @file_stat.mode
+    permission = if file_mode.to_s(8).slice(0..1) == DIRECTORY_NUMBER
+                   file_mode.to_s(8).slice(1..3).chars
+                 else
+                   file_mode.to_s(8).slice(3..5).chars
+                 end
+    permission.map { |num| PERMISSION_MAP[num] }.join
```

Comment:
```suggestion
    permission.chars.map { |num| PERMISSION_MAP[num] }.join
```

または

```suggestion
    permission.each_char.map { |num| PERMISSION_MAP[num] }.join
```

とした方が43と45行目のcharsをなくせるのでよりDRYになります

Reply:
こちらで対応しました！
https://github.com/yuki-snow1823/ruby-practices/pull/2/commits/659b9b00fc0fbd9a9a3716bc419f7d4ea67cb761